js/base/Exchange.js

```diff
@@ -1677,6 +1668,182 @@ module.exports = class Exchange {
         }
     }
 
+    getDefaultOptions () {
+        return {
+            'defaultNetworkCodeReplacements': {
+                'ETH': { 'ERC20': 'ETH' },
+                'TRX': { 'TRC20': 'TRX' },
+                'CRO': { 'CRC20': 'CRONOS' },
+            },
+            // this list is for common reserved CCXT unified network codes (some of them might have multiple aliases)
+            'unifiedNetworkCodesAndAliases': {
```

Comment:
@ttodua tbh I'm not sure if this won't do more harm than good.

First as you said, we can only return one unified network code, so the user might find it odd, for instance, doing `fetchDepositAddress('USDT', '{"network":"TRX"}')` and the response is indexed by `TRC20`.

Second, without knowing, might mess this up, because if they want to add another alias, since this is an array they  have to do something like this:

````Javascript
const currentAlias = this.options['unifiedNetworkCodesAndAliases']['USDT']
currentAlias.push("USDTTTT")
 this.options['unifiedNetworkCodesAndAliases']['USDT'] = currentAlias
```

I'm not saying it's wrong, but we need to take into consideration everything.

@kroitor your feedback here would be much appreciated

Reply:
the reason I've made `unifiedNetworkCodesAndAliases` was that CCXT users were already used to use i.e. `{networkCode: TRX}` as you mentioned, and yeah, I also prefer that we just returned exception, saying that TRX is not unified networkCode, but the thing is that CCXT already used `TRX` as acceptable alias for trc20. 
We can of course (I also vote it)  to move to only 1 networkCode for each blockchain (without any alias), but it's you @lead who should decide.

Reply:
also, `unifiedNetworkCodesAndAliases` is not meant to be used by users at all. if users are going to do some messup, then they can just literally do everything they try, i.e. `this.options = "hello world"` or whatever, but again, overriding that property will not be documented or suggested anywhere. otherwise, we don't have to fear that one user from 1000 might messup his ccxt code by doing shady things on own behalf.

Reply:
@carlosmiei @kroitor @frosty00  please share your decision, should we support networkCode aliases or not.
So, we should either support only **one** unified networkCode for each blockchain, or have aliases (so users could use `BTC` or `BITCOIN` as unified accepted code).

However, we still have to prefer one unified networkCode as "main", so users submitting `BTC` or `BITCOIN` would get i.e. `BTC` in response structure after parsing id-to-unified-code.

Reply:
so, two questions:
1) about aliases - should we support input of both BTC & BITCOIN  (and other common blockchain titles) or only one unified networkCode

2) please list the primary unified networkCodes for the following :
- for bitcoin:  `BTC` or `BITCOIN`
- for mainnet ethereum (beside ERC20):  `ETH` or `ETHEREUM`
- for solana:  `SOL` or `SOLANA`
- for tron: `TRX` or `TRON`
- for cardan: `CARDANO` or `ADA`
- for cryptocom: `CRC20` or `CRONOS`
- for dogecoin: `DOGE` or `DOGECOIN`
- for okex: `OKC` (official network name) or `OKX` (just exchange name)
- for polkadot: `POLKADOT` or `DOT`

please beware, don't get confused and prefer "currency" code for network code (i.e. `ADA` is just coin on cardano blockchain, but because of being popular coin at some degree, some exchanges mistakenly call "ada network", while it's actually cardano).

Reply:
so we should only have two 1d maps, one in the base class and one in the derived class

Reply:
@ttodua is there any exchange that uses `BITCOIN` in all caps for a network code? If yes, which exchanges? If no, then we should not introduce a `BITCOIN` alias at all.